leo-finalize.md

```diff
@@ -0,0 +1,80 @@
+
+### Leo Finalize Update
+
+The finalize programming model for on-chain code has been replaced with a new async/await model.
+
+The key difference between the two models is that instead of a finalize block, on-chain code is defined in an async function. Note that if a program does not declare on-chain code (there are not finalize blocks) then it will work without modification under the async/await model.
+
+### Previous Testnet3 Leo version Finalize model
+program test1.aleo {
+
+    //sample mapping data
+    mapping data: u32 => u32; 
+
+    //transition
+    transition main(public x: u32) -> (u32) {
+        //finalize `main` called in the finalize state below, must match with transition name
```

Comment:
I think the example could showcase a more elaborate logic were is clear that there is some off chain logic and on-chain logic that needs to take place, like updating a global state on-chain, based on some off-chain simple computation.